includes/Settings.php

```diff
@@ -287,4 +288,15 @@ private function doIterate( $key, $options ) {
 		throw new SettingNotFoundException( "'{$key}' is not a valid settings key" );
 	}
 
+	private static function initLegacyMapping( &$configuration ) {
+
+		// When adding a deprecated mapping, extend the DeprecationNoticeTaskHandler
+		// with the corresponding entry
+
+		// Deprecated in 2.5, to be removed with 3.1
```

Comment:
> Perhaps we should specify somewhere what exactly we guarantee to not break in non-breaking releases

The `Settings` legacy mapping and `DeprecationNoticeTaskHandler` allows for a more controlled process instead of just dropping the ball, so for now the process would is envisioned as:

- Remove the setting from `DefaultSettings.php` (done so to allow for a detection on whether `$GLOBALS` still contains a reference to it or not infused by the `LocalSettings.php`)
- Remap the setting in case it has been replaced by something else
- Add an entry in `DeprecationNoticeTaskHandler` to be either categorized as deprecated or removed